### PR TITLE
Replace timestamp with origin

### LINKED ISSUE #102: Replace timestamp with originTimestamp
#### Description
Right now our `Payload` objects have a `position` field (the number of ms from the stream start) and a `timestamp` field (the ISO string of the moment in time the payload is associated).  In reality, `position` and `timestamp` are conveying the same information, it's just that `position` is relative and `timestamp` is absolute.

Things get complicated because in order to calculate an absolute timestamp from a position the system needs to know the underlying "origin" timestamp.  This means that whatever portion of the system is generating timestamps is going to need to track the origin time.  Practically, accomplishing this means calculating the origin time.

This (current) approach is inefficient both in terms of computational power but also in terms of system design.  Origin timestamp calculation needs to be executed and stored in every single appliance (or via the appliance core classes) in order to later calculate timestamps.  It also means that every payload generated needs to combine {originTimestamp} and {position} (even if timestamp is never going to be used or cared about).

This issue proposes a modification to the `Payload` definition to remove `timestamp` and replace it with `originTimestamp`.  `originTimestamp` would be the same across all payloads in a given stream (unless position rolls over in which case it would update).  Creating a new payload in a stream would become trivial because it would simply copy over the originTimestamp from the input payload.

This has a huge benefit in that the system no longer has to calculate timestamps simply to generate payloads.

Appliances or applications that care about absolute time for any reason can trivially ground a payload by combining `originTimestamp` with `position` (which they might already need to do in some cases with the old model)

This change will have impact on many components of the system, but I think is a worthwhile lift and it is all better for it.

#### Related Issues
Trying to figure out the right way to calculate / support timestamp has been discussed in:

* https://github.com/tvkitchen/countertop/issues/122
* https://github.com/tvkitchen/appliances/issues/76

### PR DESCRIPTION
**Pull request by slifty**
## Description
This PR removes `timestamp` from Payload (and related classes) and replaces it with `origin`.

`timestamp` was the absolute time a payload was broadcast, while `origin` is time associated with a position value of `0`.

Ultimately this means that to calculate the absolute moment in time that a payload occurred you will need to combine `origin` with `position`.

## Related Issues
Resolves #102

### CODE REVIEW

**Comment by slifty** [comment]
Way too much time was spent thinking about whether this should be `origin` or `originAt` or `originatedAt` or something completely different.

Here are the problems with each name:

* `origin` hold a timestamp but that isn't clear from the name (as opposed to `createdAt`).  Origin could be geographic, for instance.
* `originAt` feels a little bit odd grammatically somehow.
* `originatedAt` is odd because the payload didn't *really* originate from this time, rather, this time is the origin that someone can use in order to calculate an absolute timestamp.

In short, none of the options feel perfect!

**Comment by slifty** [comment]
ORIGIN IT IS (h/t @willmoy and literally nobody else especially not @reefdog)

**Comment by reefdog** [comment]
pleased

**Comment by reefdog** [comment]
In seriousness, given the description you provided (finally!), I think `originTimestamp` would've been ideal.

IMO, the `createdAt`/`updatedAt` pattern is great 90% of the time, mostly to avoid weird semantic gymnastics around the word "timestamp", like, `creationTimestamp` or something. But 10% of the time — and yes, these percentages are statistically sound through data collection and analysis — it really is clearer just to say "this is a timestamp of thing or event X". At which point, there's no reason to contort yourself into the `At` suffix.

So, all to say: `origin` can mean _so_ many things (geographic origin? channel origin? codec origin?), that I would advocate for `originTimestamp` for clarity.

DID I HELP?

**Comment by slifty** [comment]
I AM VERY UPSET WITH YOU RIGHT NOW.

**Comment by chriszs** [comment]
Strikes me that a lot of the difficulty here is that this is being used to signal that this applies not to the object it's attached to, but a separate concept in the system. Unlike createdAt, it's not the time of the packet/payload. It applies to the stream. Therefore, what if it was `origin: { createdAt: ... }`.

**Comment by slifty** [comment]
For right now I'm going to just run with `origin` even though it isn't ideal.  At some point I'll re-visit!
